File: 125-valid-palindrome/125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    s = s.replace(/[^a-zA-Z0-9]/g, '').toLowerCase();
    var l = 0;
    var r = s.length - 1;
    console.log(s);
    while (l < r) {
        if (s[l] !== s[r]) {
            return false;
        }
        l++;
        r--;
    }
    return true;
};

File: 125-valid-palindrome/125-valid-palindrome.py

class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.replace(' ', '').lower()
        l = 0
        r = len(s) - 1
        while l < r:
            while not(s[l].isalnum()) and l < r:
                l += 1
            while not(s[r].isalnum()) and l < r:
                r -= 1
            if s[r] != s[l]:
                return False
            l += 1
            r -= 1
        return True